source/AI.h

```diff
@@ -151,7 +152,8 @@ template <class Type>
 	static Point TargetAim(const Ship &ship);
 	static Point TargetAim(const Ship &ship, const Body &target);
 	// Aim the given ship's turrets.
-	void AimTurrets(const Ship &ship, FireCommand &command, bool opportunistic = false) const;
+	void AimTurrets(const Ship &ship, FireCommand &command, bool opportunistic = false,
+			const std::optional<Point> &targetOverride = std::nullopt) const;
```

Comment:
Any particular reason why a simple pointer can't be used here? Or just a bool here and mousePosition directly in the function body?

Reply:
It's more readable if it's a strictly optional type, and it's an optional<Point> because AimTurrets isn't an player flagship-specfic thing. Maybe the individual NPCs will find a way to use this.